app/controllers/floating_ip_controller.rb

```diff
@@ -12,4 +12,262 @@ class FloatingIpController < ApplicationController
+        javascript_flash(
+          :text        => _("Floating IP creation failed: Task start failed: ID [%{id}]") % {:id => task_id.inspect},
+          :severity    => :error,
+          :spinner_off => true
+        )
+      end
+    end
+  end
+
+  def create_finished
+    task_id = session[:async][:params][:task_id]
+    floating_ip_name = session[:async][:params][:name]
+    task = MiqTask.find(task_id)
+    if MiqTask.status_ok?(task.status)
+      add_flash(_("Floating IP \"%{name}\" created") % { :name  => floating_ip_name })
+    else
+      add_flash(
+        _("Unable to create Floating IP \"%{name}\": %{details}") % { :name    => floating_ip_name,
+                                                                      :details => task.message }, :error)
+    end
+
+    @breadcrumbs.pop if @breadcrumbs
+    session[:edit] = nil
+    session[:flash_msgs] = @flash_array.dup if @flash_array
+
+    javascript_redirect :action => "show_list"
+  end
+
+  def delete_floating_ips
+    assert_privileges("floating_ip_delete")
+
+    floating_ips = if @lastaction == "show_list" || (@lastaction == "show" && @layout != "floating_ip")
+                     find_checked_items
+                   else
+                     [params[:id]]
+                   end
+
+    if floating_ips.empty?
+      add_flash(_("No Floating IPs were selected for deletion."), :error)
+    end
+
+    floating_ips_to_delete = []
+    floating_ips.each do |s|
+      floating_ip = FloatingIp.find_by_id(s)
+      if floating_ip.nil?
+        add_flash(_("Floating IP no longer exists."), :error)
+      else
+        floating_ips_to_delete.push(floating_ip)
+      end
+    end
+    process_floating_ips(floating_ips_to_delete, "destroy") unless floating_ips_to_delete.empty?
+
+    # refresh the list if applicable
+    if @lastaction == "show_list"
+      show_list
+      @refresh_partial = "layouts/gtl"
+    elsif @lastaction == "show" && @layout == "floating_ip"
+      @single_delete = true unless flash_errors?
+      if @flash_array.nil?
+        add_flash(_("The selected Floating IP was deleted"))
+      end
+    end
+  end
+
+  def edit
+    assert_privileges("floating_ip_edit")
+    @floating_ip = find_by_id_filtered(FloatingIp, params[:id])
+    @in_a_form = true
+    drop_breadcrumb(
+      :name => _("Associate Floating IP \"%{name}\"") % { :name  => @floating_ip.name },
+      :url  => "/floating_ip/edit/#{@floating_ip.id}git")
+  end
+
+  def floating_ip_form_fields
+    assert_privileges("floating_ip_edit")
+    # TODO: find_by_id_filtered fails
+    # floating_ip = find_by_id_filtered(FloatingIp, params[:id])
+    floating_ip = FloatingIp.find(params[:id])
+    network_port_ems_ref = if floating_ip.network_port
+                             floating_ip.network_port.ems_ref
+                           else
+                             ""
+                           end
+    # TODO: router field is missing!
+    # :router_id            => floating_ip.router.id,
```

Comment:
I don't think you can really change a router_id? It should be router used when you connect a floating IP to the VM. So you can't change it directly. (At least I don;t see that option in API)